packages/vm/src/evm/opcodes/util.ts

```diff
@@ -258,6 +263,42 @@ export function updateSstoreGas(
       -> Value is zero, but slot is nonzero
       Thus, the remaining case is where value is nonzero, but slot is zero, which is this clause
     */
-    return new BN(common.param('gasPrices', 'sstoreSet'))
+    return BigInt(common.param('gasPrices', 'sstoreSet'))
+  }
+}
+
+export function mod(a: bigint, b: bigint) {
+  let r = a % b
+  if (r < BigInt(0)) {
+    r = b + r
+  }
+  return r
+}
+
+export function fromTwos(a: bigint) {
+  return BigInt.asIntN(256, a)
+}
+
+export function toTwos(a: bigint) {
+  return BigInt.asUintN(256, a)
+}
+
+export function abs(a: bigint) {
+  if (a > 0) {
+    return a
+  }
+  return a * BigInt(-1)
+}
+
+const N = BigInt(115792089237316195423570985008687907853269984665640564039457584007913129639936)
+export function exponentation(bas: bigint, exp: bigint) {
```

Comment:
Is this `modExp` again? Why two duplicate methods? Also I think it makes sense to use `require('ethereum-cryptography').CURVE.n` if vm has access to it somehow.

Reply:
It's not intended to be.  I followed the 2^k-ary method [here](https://www.geeksforgeeks.org/exponential-squaring-fast-modulo-multiplication/) for doing exponentiation.  And the `vm` package doesn't currently have `ethereum-cryptography` as a dependency so we don't currently have access to the constant you referenced.  We could add it for develop but I'm not sure if it makes sense to add a new dependency just for that.  If/when we pull the `ethereum-cryptography` re-exports out of `util` and reference them directly, I imagine we will pull that dependency into the VM and could then reference it there.